src/transformStateWithClones.js

```diff
@@ -7,7 +7,42 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const results = [];
+  let temporaryPreviousObject = { ...state };
+
+  for (const item of actions) {
+    const type = item.type;
+
+    switch (type) {
+      case 'addProperties':
+        const addPropertiesObj = {
+          ...temporaryPreviousObject,
+          ...item.extraData,
+        };
+
+        temporaryPreviousObject = addPropertiesObj;
+
+        results.push(addPropertiesObj);
+        break;
+      case 'removeProperties':
+        const removePropertiesObj = { ...temporaryPreviousObject };
+
+        for (const key of item.keysToRemove) {
+          delete removePropertiesObj[key];
+        }
+
+        temporaryPreviousObject = removePropertiesObj;
+
+        results.push(removePropertiesObj);
```

Comment:
you can just delete from `temporaryPreviousObject`

Reply:
Thanks, I fixed it.